src/gsi/read_radarref_mosaic_directDA.f90

```diff
@@ -0,0 +1,490 @@
+subroutine read_radarref_directDA(nread,ndata,nodata,infile,obstype,lunout,twind,sis,nobs)
+!$$$  subprogram documentation block
+!                .      .    .                                       .
+! subprogram:  read_radarref_directDA    Reading in reflectivity mosaic in RR grid
+!
+!   PRGMMR: Ming Hu          ORG: NP22        DATE: 2006-03-27
+!
+! ABSTRACT: 
+!     This routine read in reflectivity mosaic data.  The data has already
+!          been interpolated into analysis grid and in form of BUFR.
+!
+! PROGRAM HISTORY LOG:
+!    2008-12-20  Hu  make it read in BUFR form reflectivity  data
+!    2010-04-09  Hu  make changes based on current trunk style
+!    2013-03-27  Hu  add code to map obs from WRF mass H grid to analysis grid
+!    2015-02-23  Rancic/Thomas - add l4densvar to time window logical
+!    2015-03-23  Su  -fix array size with maximum message and subset number from
+!                        fixed number to dynamic allocated array
+!    2016-09-xx  CAPS(G. Zhao) 
+!                     - modified Ming Hu's read_radarref_mosaic for 
+!                     - reading mosaic ref dbz for GSIv3.5 variational and
+!                     - enkf analysis, not for gsd-cloud-analysis
+!    2016-11-xx  CAPS(G. Zhao)
+!                     - fix the bugs:
+!                     - dbz bufr with scale=2 to have two more digits accuracy
+!                     - station id
+!                     - obs lat/lon in degree
+!
+!   input argument list:
+!     infile   - unit from which to read mosaic information file
+!     obstype  - observation type to process
+!     lunout   - unit to which to write data for further processing
+!     twind    - input group time window (hours)
+!     sis      - observation variable name
+!
+!   output argument list:
+!     nread    - number of type "obstype" observations read
+!     ndata    - number of type "obstype" observations retained for further processing
+!     nobs     - array of observations on each subdomain for each processor
+!
+! USAGE:
+!   INPUT FILES:  refInGSI
+!
+!   OUTPUT FILES:
+!
+! REMARKS:
+!
+! ATTRIBUTES:
+!   LANGUAGE: FORTRAN 90
+!   MACHINE:  Linux cluster(Wjet)
+!
+!$$$
+!
+!_____________________________________________________________________
+!
+  use kinds, only: r_kind,r_double,i_kind
+  use constants, only: zero,one,rad2deg,deg2rad
+  use convinfo, only: nconvtype,ctwind,cgross,cermax,cermin,cvar_b,cvar_pg, &
+        ncmiter,ncgroup,ncnumgrp,icuse,ictype,icsubtype,ioctype
```

Comment:
Variables cgross, cermax, cermin, cvar_b, cvar_pg, ncmiter, ncgroup, ncnumgrp, ictype, and icsubtype have been identified as unused.  Please remove unused variables.

Reply:
We removed them as you suggested.

https://github.com/NOAA-EMC/GSI/pull/142/files#diff-1833e300b7dd38188cb415090239f3f184b0ad14deaae73fb732e4fa73d4f221R58